### PR TITLE
New module displaying the address book of your employees in the website

### PR DESCRIPTION
**Pull request by lmignon**

### CODE REVIEW

File: website_hr_contact/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        sudo_hr_department = hr_department.sudo()
+        visible_departments = hr_department.search([(1, '=', 1)])
+        departments = sudo_hr_department.search([('parent_id', '=', False)])
+        for department in departments:
+            children = self._get_children_department_recursive(
+                department, visible_departments)
+            if department in visible_departments:
+                tree.append({
+                    'department': visible_departments.filtered(
+                        lambda rec: rec.id == department.id),
```
Comment (andreparames) [inline website_hr_contact/controllers/main.py:98]: Isn't this equivalent to `'department': department,`?

Comment (lmignon) [inline website_hr_contact/controllers/main.py]: @andreparames good catch

Comment (lmignon) [inline website_hr_contact/controllers/main.py:98]: @andreparames `department` is a department in the `sudo`Environment. By filtering visible_departments I take the department in the user's Environement.

Comment (andreparames) [inline website_hr_contact/controllers/main.py:98]: Right, sorry, I missed the sudo.

**Comment by rafaelbn** [comment]
@lmignon which is the difference with https://github.com/odoo/odoo/tree/8.0/addons/website_hr ?

**Comment by rafaelbn** [comment]
```
2016-06-03 22:57:15     CRITICAL    server  openerp.service.server:949 preload_registries

Failed to initialize database `3148241-82-80e670-all`.
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/service/server.py", line 936, in preload_registries
    registry = RegistryManager.new(dbname, update_module=update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/modules/registry.py", line 370, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/modules/loading.py", line 354, in load_modules
    loaded_modules, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/modules/loading.py", line 255, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/modules/loading.py", line 179, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='demo')
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/modules/loading.py", line 118, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/tools/convert.py", line 900, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/tools/convert.py", line 986, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/tools/convert.py", line 852, in parse
    self._tags[rec.tag](self.cr, rec, n, mode=mode)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/tools/convert.py", line 762, in _tag_record
    id = self.pool['ir.model.data']._update(cr, self.uid, rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode, context=rec_context )
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/addons/base/ir/ir_model.py", line 1066, in _update
    res_id = model_obj.create(cr, uid, values, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/api.py", line 372, in old_api
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/addons/website_slides/models/slides.py", line 434, in create
    values['url']).get('values', dict())
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/addons/website_slides/models/slides.py", line 581, in _parse_document_url
    document_id, only_preview_fields)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/addons/website_slides/models/slides.py", line 595, in _parse_youtube_document
    youtube_values = fetch_res['values'].get('items', list(dict()))[0]
ParseError: "list index out of range" while parsing /srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3148241-82-80e670/openerp/addons/website_slides/data/website_slides_demo.xml:76, near
<record id="slide_5" model="slide.slide">
            <field name="name">OpenERP POS Indiegogo Campaign</field>
            <field name="url">https://www.youtube.com/watch?v=C_Laaz4mORg</field>
            <field name="channel_id" ref="channel_public"/>
            <field name="category_id" ref="category_1"/>
            <field name="website_published" eval="True"/>
            <field name="tag_ids" eval="[(4, ref('tag3'))]"/>
        </record>
```

**Comment by rafaelbn** [comment]
Hi @lmignon if you answer me we can go ahead with this PR and get it merged. Let me know. Thanks!

**Comment by lmignon** [comment]
Hi @rafaelbn  https://github.com/odoo/odoo/tree/8.0/addons/website_hr display a flat list of your employees without their addresses. This module let's you display a list of employees with their addresses grouped by department.

**Comment by rafaelbn** [comment]
Thanks @lmignon . Could you check runbot please?

**Review by JonathanNEMRY — APPROVED** [review]

**Comment by JonathanNEMRY** [comment]
For me it is ready to merge as @lmignon did all demanded fixes and as far as I can see it the pr is green :smile: 
some other :+1: 
cc @adrienpeiffer  @rafaelbn

**Comment by JonathanNEMRY** [comment]
cc @andreparames

**Comment by andreparames** [comment]
Seems fine to me :+1:

**Review by rafaelbn — APPROVED** [review]
Tested in runbot

**Pull request merged**